### PR TITLE
[Merged by Bors] - fix(Testing/SlimCheck/Testable): do not call `inferType` on an expression with `bvar`s

### PR DESCRIPTION
**Pull request by eric-wieser**
Fixes a bug reported twice on Zulip:
* https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/slim_check.20question/near/412709012
* https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/slim_check.20giving.20wrong.20counterexamples.3F/near/420008365

Both test-cases are included

---

- [ ] depends on: #10393

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Testing/SlimCheck/Testable.lean [hunk 1]
```diff
@@ -512,11 +512,11 @@ partial def addDecorations (e : Expr) : MetaM Expr :=
     if not (← Meta.inferType e).isProp then
       return .continue
     else if let Expr.forallE name type body data := expr then
-      let n := name.toString
       let newType ← addDecorations type
-      let newBody ← addDecorations body
+      let newBody ← Meta.withLocalDecl name data type fun fvar => do
+        return (← addDecorations (body.instantiate1 fvar)).abstract #[fvar]
```
Comment (eric-wieser) [inline Mathlib/Testing/SlimCheck/Testable.lean:517]: I'm surprised that there is no `abstract1` here; I'm also surprised that there don't seem to be any other users of `abstract` in mathlib.

Comment (digama0) [reply Mathlib/Testing/SlimCheck/Testable.lean:517]: normally the abstract is folded together with the construction of the forall using `mkForall`

Comment (eric-wieser) [reply Mathlib/Testing/SlimCheck/Testable.lean:517]: Are you advocating a change here?

Comment (eric-wieser) [reply Mathlib/Testing/SlimCheck/Testable.lean:517]: The documentation for `mkForall` is
```
/--
`.forallE x t b bi` is now the preferred form.
-/
def mkForall (x : Name) (bi : BinderInfo) (t : Expr) (b : Expr) : Expr :=
  .forallE x t b bi
```
which I assume should be read as a deprecation?

Comment (digama0) [reply Mathlib/Testing/SlimCheck/Testable.lean:517]: sorry, I meant `mkForallFVars`

**Comment by semorrison** [comment]
bors d+

File: Mathlib/Testing/SlimCheck/Testable.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/Testing/SlimCheck/Testable.lean:517]: Are you happy with this as is, or do you want me to try to rewrite to use `mkForallFVars`?

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**